src/defs.h

```diff
@@ -536,53 +536,43 @@ namespace std {
 #if !defined(WIN32) && !defined(__FreeBSD__)
 namespace tr1 {
 #endif
-template <> struct hash<struct sockaddr_in> : public std::unary_function<struct sockaddr_in, int> {
-    int operator()(struct sockaddr_in const &key) const {
-        // XOR "a.b" part of "a.b.c.d" address with 16bit port; leave "c.d" part untouched for
-        // maximum hashing
-        return key.sin_addr.s_addr ^ key.sin_port;
-    }
-};
-
-template <>
-struct hash<struct port_and_type> : public std::unary_function<struct port_and_type, int> {
-    int operator()(struct port_and_type const &key) const {
+template <> struct hash<struct sockaddr_store_t> : public std::unary_function<struct sockaddr_store_t, int> {
+    int operator()(struct sockaddr_store_t const &key) const {
         // XOR "a.b" part of "a.b.c.d" address with 16bit port; leave "c.d" part untouched for
         // maximum hashing
-        return key.sock_type ^ key.port;
+        // use only IPv4 part for hashing
+        return ((sockaddr_in &)key).sin_addr.s_addr ^ ((sockaddr_in &)key).sin_port;
```

Comment:
Would this be enough for IPv6?

Reply:
Added other hash for IPv6. Made IPv4 hashing correspond to comment.